cedar-policy-core/src/evaluator.rs

```diff
@@ -252,9 +252,383 @@ impl<'e> Evaluator<'e> {
+                                }),
+                            ..
+                        },
+                        BinaryOp::ContainsAny,
+                    ) => Ok((arg2_set.is_disjoint(&arg1_set)).into()),
+                    (
+                        Value {
+                            value: ValueKind::Set(arg1_set),
+                            ..
+                        },
+                        Value {
+                            value: ValueKind::Set(arg2_set),
+                            ..
+                        },
+                        BinaryOp::ContainsAll,
+                    ) => Ok(arg2_set
+                        .authoritative
+                        .iter()
+                        .all(|item| arg1_set.authoritative.contains(item))
+                        .into()),
+                    (
+                        Value {
+                            value: ValueKind::Set(arg1_set),
+                            ..
+                        },
+                        Value {
+                            value: ValueKind::Set(arg2_set),
+                            ..
+                        },
+                        BinaryOp::ContainsAny,
+                    ) => Ok(arg1_set
+                        .authoritative
+                        .iter()
+                        .any(|item| arg2_set.authoritative.contains(item))
+                        .into()),
+                    (_, _, BinaryOp::ContainsAll) | (_, _, BinaryOp::ContainsAny) => {
+                        let culprit = if arg1.get_as_set().is_err() {
+                            arg1
+                        } else {
+                            arg2
+                        };
+                        Err(EvaluationError::type_error_with_advice(
+                            nonempty![Type::Set],
+                            &culprit,
+                            format!("operation `{op}` should have set operands"),
+                        ))
+                    }
+                }
+            }
+            ExprKind::MulByConst { arg, constant } => {
+                let arg = self.interpret(&arg, slots)?;
+                arg.get_as_long()?
+                    .checked_mul(*constant)
+                    .map(Value::from)
+                    .ok_or_else(|| {
+                        EvaluationError::integer_overflow(
+                            IntegerOverflowError::Multiplication {
+                                arg,
+                                constant: *constant,
+                            },
+                            loc.cloned(),
+                        )
+                    })
+            }
+            ExprKind::ExtensionFunctionApp { fn_name, args } => {
+                let efunc = self
+                    .extensions
+                    .func(&fn_name)
+                    .map_err(|err| EvaluationError::extension_function_lookup(err, loc.cloned()))?;
```

Comment:
I think this will be a behavior change vs the current partial evaluator.  The current partial evaluator evaluates all the arguments first, before checking if the efunc lookup fails.  So if an argument has an error _and_ the efunc lookup is invalid, the current partial evaluator will prefer the argument error, while this code will prefer the efunc lookup error.  I don't think we have a strong reason to prefer one over the other, and I don't think we mind that a different error is thrown now (I think?), but just flagging this